HIP/hip-412.md

```diff
@@ -213,7 +233,149 @@ Note that mime types for directories are not uniformly defined. Some IPFS CIDs p
 
 ## Reference Implementation
 
-#### Example Schema: none
+### Example Schema: Collectibe Hedera NFTs (format: "HIP412")
+
+This is a **recommended reference implementation for collectible Hedera NFTs**. The `HIP412` standard has been designed to be used by all NFT tooling (wallets, explorers) and be mostly compatible with other existing standards. 
+
+Here's an example of a full implementation of the metadata schema described in this `HIP412` specification for an image-based NFT. We are setting the image field to a URI and including the `sha256_checksum` field which represents a hash of the provided image. 
+
+The `files` array contains the main image hosted on IPFS, also including an optional checksum for validation purposes. The `image` field serves as a preview image for NFT tooling where a programatically generated image doesn't work or you would like to have a different preview image. Therefore, the `image` field remains optional.
+
+Further, no additional properties are defined on the JSON object on root level. Any additional properties you want to define should go into the `properties` object.
+
+You can optionally define `attributes` to calculate rarity scores. We've added the field `display_type` similar to the OpenSea standard which defines how the attribute should be displayed.
+
+The standard also allows for localization. Each locale links to another metadata file that contains the localized metadata and files. This allows for a clean metadata structure. Don't define another localization object for a localized metadata file to avoid infinite looping when parsing an NFT's metadata file.
+
+**A JSON schema can be found [here](https://gist.github.com/michielmulders/1565e7759d692b65fa8c38cfe15b010c).**
+
+```json
+{
+	"name": "Example NFT 001",
+	"creator": "Jane Doe, John Doe",
+	"creatorDID": "did:hedera:mainnet:7Prd74ry1Uct87nZqL3ny7aR7Cg46JamVbJgk8azVgUm;hedera:mainnet:fid=0.0.123",
+	"description": "This describes my NFT",
+	"image": "https://myserver.com/nft-001.png",
+	"sha256_checksum": "ba7816bf8f01cfea414140de5dae2223b00361a396177a9cb410ff61f20015ad",
+	"type": "image/png",
+	"format": "HIP412",
```

Comment:
I understand that any format could be added, but do you think we should list the baseline formats that can be used?

So far in this doc I've seen:
`none`
`opensea`
`HIP412`

Also, as a parser of this metadata, how would I know which version of HIP412 the file is?

Should we consider adding version to the format as the HIP adapts over time?

Something like:
```
HIP412-2022-04-27
HIP412-2022-09-20
```
or
```
HIP412@1.0.0
HIP412@1.0.1
```

Reply:
Here's a good blog on how to use semantic versioning for data schemas https://snowplow.io/blog/introducing-schemaver-for-semantic-versioning-of-schemas/